### PR TITLE
Posthook fixes

### PR DESCRIPTION
**Pull request by bonifaido**
- Make StoreKubeConfig posthook function idempotent
- Log posthook states as status message during cluster creation

### CODE REVIEW

File: cluster/hooks.go [hunk 1]
```diff
@@ -23,17 +23,23 @@ import (
...
-				i.Error(createdCluster, err)
+				log.Errorf("Error during posthook function[%s]: %s", postHook, err.Error())
+				postHook.Error(createdCluster, err)
 				return err
 			}
+
+			statusMsg := fmt.Sprintf("Posthook function finished: %s", postHook)
+			err = createdCluster.UpdateStatus(pkgCluster.Creating, statusMsg)
+			if err != nil {
+				log.Errorf("Error during posthook status update in db [%s]: %s", postHook, err.Error())
```
Comment (baluchicken) [inline cluster/hooks.go]: Shouldn't we return here with the error? Otherwise we will have inconsistent state in the DB

File: secret/store.go [hunk 2]
```diff
@@ -27,6 +27,9 @@ var log *logrus.Logger
 // Store object that wraps up vault logical store
 var Store *secretStore
 
+// ErrSecretNotExists denotes 'Not Found' errors for secrets
+var ErrSecretNotExists = fmt.Errorf("There's no secret with this ID")
```
Comment (baluchicken) [inline secret/store.go]: This can be a const

File: cluster/hooks.go (hunk 1 above)
Comment (bonifaido) [reply cluster/hooks.go]: Right, I will return the err. Thanks!

File: secret/store.go (hunk 2 above)
Comment (baluchicken) [reply secret/store.go]: My bad sorry this is a struct:)

Comment (bonifaido) [reply secret/store.go]: 👍

**Pull request merged**